tests/plugins/inlinetoolbar/context/refresh.js

```diff
@@ -0,0 +1,67 @@
+/* bender-tags: inlinetoolbar,context */
+/* bender-ckeditor-plugins: inlinetoolbar */
+
+( function() {
+	'use strict';
+
+	bender.editor = {
+		config: {}
+	};
+
+	bender.test( {
+		setUp: function() {
+			if ( CKEDITOR.env.ie && CKEDITOR.env.version === 8 ) {
+				assert.ignore();
+			}
+		},
+
+		tearDown: function() {
+			this.editor.inlineToolbar._manager._clear();
+		},
+
+		'test refresh returning true': function() {
+			var context = this._getContextStub( sinon.stub().returns( true ), true );
+
+			this._assertToolbarVisible( true, context );
+		},
+
+		'test refresh returning false': function() {
+			var context = this._getContextStub( sinon.stub().returns( false ), true );
+
+			this._assertToolbarVisible( false, context );
+		},
+
+		'test refresh returning falsy value': function() {
+			var context = this._getContextStub( sinon.stub().returns( 0 ), true );
+
+			this._assertToolbarVisible( false, context );
+		},
+
+		/*
+		 * Returns a Context instance with toolbar show/hide methods stubbed.
+		 *
+		 * @param {Function} refreshCallback Function to be used as `options.refresh`.
+		 * @param {Boolean} [autoRefresh=false] Whether function should automatically force context
+		 * manager, to recheck all the contexts.
+		 * @returns {CKEDITOR.plugins.inlinetoolbar.context}
+		 */
+		_getContextStub: function( refreshCallback, autoRefresh ) {
```

Comment:
To be honest, each function like that is slightly different, and after making a common interface it would not be as usable for each suite.

Thus I don't see too much value in doing that. WDYT?

Reply:
I see, you would have to make generic function which can take `refresh`, `widgets` and `cssSelector` and create specific context based on that. As it will overcomplicate thinks, lets leave it as is 👍